common/utils/src/main/java/org/apache/spark/internal/Logger.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import scala.collection.immutable.Seq;
+import scala.jdk.javaapi.CollectionConverters;
+
+import org.apache.logging.log4j.CloseableThreadContext;
+
+public class Logger {
+
+  private final org.slf4j.Logger slf4jLogger;
+
+  Logger(org.slf4j.Logger slf4jLogger) {
+    this.slf4jLogger = slf4jLogger;
+  }
+
+  public void error(String msg) {
+    if (slf4jLogger.isErrorEnabled()) {
+      slf4jLogger.error(msg);
+    }
+  }
+
+  public void error(String msg, Throwable throwable) {
+    if (slf4jLogger.isErrorEnabled()) {
+      slf4jLogger.error(msg, throwable);
+    }
+  }
+
+  public void error(String msg, MDC... mdcs) {
```

Comment:
Although the following design is shown in the document, 
<img width="622" alt="image" src="https://github.com/apache/spark/assets/15246973/0dce3f40-6200-40fe-bbd8-459083a10f30">
But I think the parameter `void error(String msg, MDC... mdcs)` is more in line with the style of Java? WDYT @gengliangwang

Reply:
Either way is fine to me